lib/hodor/environment.rb

```diff
@@ -241,6 +242,48 @@ def ssh_addr
       va << " -p #{settings[:ssh_port] || 22}" 
     end
 
+    def yml_expand(val, parents)
+      if val.is_a? String
+        val.gsub(/\$\{.+?\}/) { |match|
+          cv = match.split(/\${|}/)
+          expr = cv[1]
+          ups = expr.split('^')
+          parent_index = parents.length - ups.length
+          parent = parents[parent_index]
+          parent_key = ups[-1]
+          parent_key = parent_key[1..-1] if parent_key.start_with?(':')
+          if parent.has_key?(parent_key)
+            parent[parent_key]
+          elsif parent.has_key?(parent_key.to_sym)
+            parent[parent_key.to_sym]
+          else
+            parent_key
+          end
+        }
+      elsif val.is_a? Hash
+        more_parents = parents << val
+        val.each_pair { |k, v|
+          exp_val = yml_expand(v, more_parents)
+          val[k] = exp_val
+        }
+      else
+        val
+      end
+    end
+
+    def yml_flatten(parent_key, val)
```

Comment:
Is yml_flatten being used anywhere yet? I had originally referenced it within one of the job.properties files.